### ISSUE TITLE
onClick Event on anchor tag not possible.? Custom button on the table not possible?

### ISSUE THREAD

**Issue by onkar-joshi-invimatic** [akveo/ng2-smart-table#348]
I am using the ng2-smart-table. I want to add a new button to All my columns in the table. It seems it is not possible in any way. So i looked through some issues and found out people are using custom anchor tags which is apparently possible. I tried using the anchor tag and the href property does take me to a new page. But i dont want to move to any other page i want to raise a function from my component onClick of either the button or the anchor tag. Is this going to be possible? If so, please help me out. 

Summing up in simple words: 
1) How to add a custom button to the smart table.?
2) How to assign onclick event to an anchor (<a>) tag.?

**Comment by arun-awnics** [comment]
Hopefully this should help you: https://plnkr.co/edit/lGP4XsO64gjgnTg9Qpvn?p=preview

**Comment by onkar-joshi-invimatic** [comment]
Yes that did help me thank you for that. But yet another problem i am facing since yesterday. I am trying to use the 'append' function provided in the documentation to 'append' the new row to the table and not 'prepend' it as it is doing by default. Everytime i add a new row it gets added "ABOVE" the last entered row which is not the usual activity i expect in my table. I want the new row to be added to the next line. 
So, for this, they have provided an "APPEND" method which is mentioned in their documentation but there is NO reference to any resource which tells us HOW exactly we are supposed to use/implement these methods. If you can look into this one it would be a great help. Thank You.

**Comment by arun-awnics** [comment]
Hi @onkar-joshi-invimatic `this.source.add(object)` behaves like `this.source.append(object)` by default. If you want the row to be added in the beginning of the table then you have to use `this.source.prepend(object)`. Kindly check the plunkr link I have shared above to see the demonstration of the same. You can try changing the `add(), append() and prepend()` in create-modal.component.ts file to see the changes.
